spec.html

```diff
@@ -29575,32 +29576,41 @@ <h1>Disjunction</h1>
       <!-- es6num="21.2.2.4" -->
       <emu-clause id="sec-alternative">
         <h1>Alternative</h1>
+        <p>With argument _direction_.</p>
         <p>The production <emu-grammar>Alternative :: [empty]</emu-grammar> evaluates as follows:</p>
         <emu-alg>
           1. Return a Matcher that takes two arguments, a State _x_ and a Continuation _c_, and returns the result of calling _c_(_x_).
         </emu-alg>
         <p>The production <emu-grammar>Alternative :: Alternative Term</emu-grammar> evaluates as follows:</p>
         <emu-alg>
-          1. Evaluate |Alternative| to obtain a Matcher _m1_.
-          1. Evaluate |Term| to obtain a Matcher _m2_.
-          1. Return an internal Matcher closure that takes two arguments, a State _x_ and a Continuation _c_, and performs the following steps when evaluated:
-            1. Let _d_ be a Continuation that takes a State argument _y_ and returns the result of calling _m2_(_y_, _c_).
-            1. Call _m1_(_x_, _d_) and return its result.
+          1. Evaluate |Alternative| with argument _direction_ to obtain a Matcher _m1_.
+          1. Evaluate |Term| with argument _direction_ to obtain a Matcher _m2_.
+          1. If _direction_ is equal to +1, then
+            1. Return an internal Matcher closure that takes two arguments, a State _x_ and a Continuation _c_, and performs the following steps when evaluated:
+              1. Let _d_ be a Continuation that takes a State argument _y_ and returns the result of calling _m2_(_y_, _c_).
+              1. Call _m1_(_x_, _d_) and return its result.
+          1. Else,
+            1. Assert: _direction_ is equal to -1.
+            1. Return an internal Matcher closure that takes two arguments, a State _x_ and a Continuation _c_, and performs the following steps when evaluated:
+              1. Let _d_ be a Continuation that takes a State argument _y_ and returns the result of calling _m1_(_y_, _c_)
+              1. Call _m2_(_x_, _d_) and return its result.
         </emu-alg>
         <emu-note>
-          <p>Consecutive |Term|s try to simultaneously match consecutive portions of _Input_. If the left |Alternative|, the right |Term|, and the sequel of the regular expression all have choice points, all choices in the sequel are tried before moving on to the next choice in the right |Term|, and all choices in the right |Term| are tried before moving on to the next choice in the left |Alternative|.</p>
+          <p>Consecutive |Term|s try to simultaneously match consecutive portions of _Input_. When _direction_ is equal to +1, if the left |Alternative|, the right |Term|, and the sequel of the regular expression all have choice points, all choices in the sequel are tried before moving on to the next choice in the right |Term|, and all choices in the right |Term| are tried before moving on to the next choice in the left |Alternative|. When _direction_ is equal to -1, the evaluation order of |Alternative| and |Term| are reversed.</p>
```

Comment:
is it clear here to use "+1" and "-1", and then also to use "left" and "right", which might not be correct in an RTL context?

Reply:
IMHO it’s clear that this refers to the grammar productions, which are always in a left-to-right order.

Reply:
Instead of -1/+1, what would you think about using the strings "left" and "right"? (obv they're merely spec artifacts either way)

Reply:
Do you think that would be more clear? I’m in favor of making things more readable but personally don’t see how this change does that.

Reply:
@claudepache This goes back to @ljharb’s first comment. IMHO it’s clear that directionality of the actual characters doesn’t play a role as we’re just talking about grammar productions here. Do you disagree?

Reply:
- I think that -1/+1 is confusing when the value isn't used directly in a math operation
 - i think "left" and "right" can be problematic and confusing, even when it makes sense (as we've seen with pad/trim/reduce)
 - I don't have better words for the direction argument in mind :-/ maybe "advance" (to go with AdvanceStringIndex) and "regress" (or some other antonym for "advance")?

Reply:
@ljharb Maybe `backwards` and `forwards`? Similar to the usage in the `Intl.Segmenter` proposal.

Reply:
Wow, that's way simpler - yeah that would make it much clearer for me.

(note that i'm not trying to block anything; just sharing my thoughts prior to merging. I wouldn't want to delay this landing :-) )

Reply:
@littledan if a regex is in unicode mode, it might advance +2 when it finds a surrogate pair, no?

Reply:
(It'll only be advancing +-1 code point...) I don't feel strongly about this. I'm fine with switching to `~forwards~` and `~backwards~`.